cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerHttpHandlerInternal.java

```diff
@@ -87,6 +100,79 @@ public void run(FullHttpRequest request, HttpResponder responder) {
     }
   }
 
+  /**
+   * Run a task specified by query parameters: {@code className} and {@code param}
+   *
+   * @param request http request whose body contains {@link RunnableTaskRequest} in json format.
+   * @param responder {@link HttpResponder}
+   */
+
+  /**
+   * Run a task specified by query parameters {@code className} and {@code param}.
+   * <p>
+   * Request body contains the content of a file uploaded by the caller that is available to the task
+   * at runtime via {@link RunnableTaskContext}.
+   *
+   * @param request {@link HttpRequest} whose body should be a file content.
+   * @param responder {@link HttpResponder}
+   * @param className classname of the task to run
+   * @param param parameter(s) to the task
+   * @return {@link BodyConsumer}
+   */
+  @POST
+  @Path("/task")
```

Comment:
nit: /run and /task might be a bit confusing. I though don't have a better suggestion either.